### ISSUE TITLE
Very blurry icons in Retina screen

### ISSUE THREAD

**Issue by memeplex** [emacs-lsp/lsp-mode#3143]
### Thank you for the bug report

- [X] I am using the latest version of `lsp-mode` related packages.
- [X] I checked [FAQ](https://emacs-lsp.github.io/lsp-mode/page/faq/) and [Troubleshooting](https://emacs-lsp.github.io/lsp-mode/page/troubleshooting/) sections
- [X] You may also try reproduce the issue using clean environment using the following command
```bash
emacs -q -l lsp-start-plain.el
```
where `lsp-start-plain.el` can be downloaded [here](https://github.com/emacs-lsp/lsp-mode/blob/master/scripts/lsp-start-plain.el).

### Bug description

All icons look blurry like this one:

![image](https://user-images.githubusercontent.com/2845433/136671764-a2063b2d-90f4-48fe-9246-36c602934f1f.png)

Presumably they are rendered as for a 96 dpi screen and then scaled up x 2.

I'm using emacs-28 branch built from source with svg support. `(frame-scale-factor)` is correctly reported as 2. My laptop is a MacBook Pro 2017.

### Steps to reproduce

Just install lsp-mode in a MacBook using the official ns/cocoa emacs port and look anywhere a icon is (for example, the breadcrumb).

### Expected behavior

Icons are crisp in a HiDPI screen.

### Which Language Server did you use?

Not relevant, but lsp-pyright.

### OS

MacOS

### Anything else?

I should probably report this to treemacs also. But I'm not sure whether treemacs is an upstream in this regard (font rendering).

**Comment by yyoncho** [comment]
We should switch to svg for these icons.

**Comment by memeplex** [comment]
That would be better for sure, but beware that some part of emacs still ignore scale factors, see https://debbugs.gnu.org/cgi/bugreport.cgi?bug=50958.

**Comment by yyoncho** [comment]
As a side note, are the icons in the completion menu working fine?

**Comment by memeplex** [comment]
No, svg is likely the right way, I'm just giving a heads up, maybe the issue is only related to docview but svg rendering in general will work OOTB at the right resolution.

**Comment by yyoncho** [comment]
you know that built-in company-mode already supports icons and also it has two themes - one for dart themes and one for light. ATM you are looking at the images for dark theme
